plugins/UsersManager/API.php

```diff
@@ -783,16 +783,13 @@ private function isUserTheOnlyUserHavingSuperUserAccess($userLogin)
      * Generates a unique MD5 for the given login & password
      *
      * @param string $userLogin Login
-     * @param string $md5Password MD5ied string of the password
-     * @throws Exception
+     * @param string $passwordHash hashed string of the password
      * @return string
      */
-    public function getTokenAuth($userLogin, $md5Password)
+    public function getTokenAuth($userLogin, $passwordHash)
```

Comment:
That's why tests fail as it breaks the API. Unfortunately we cannot rename this one although it would be good :( 

Users call it eg with `index.php?module=API&method=API.getTokenAuth&userLogin=foo&md5Password=bar...`. This would break all applications, libraries etc that use this method (eg the Piwik Mobile App but also many other libraries and apps).

Unfortunately we do not have a versioning in our API to handle such cases. The only thing that I could imagine would be making `$md5Password` optional, add `$passwordHash` optional and then checking in the method that at least one of them is set and if not throw an exception.

We'd need to document this in the developer changelog and deprecate `$md5Password` parameter. We'd probably never actually remove that parameter instead deprecate the whole API and replace it with a REST API in a few years